docs/source/examples/mg-mode-part-i.ipynb

```diff
@@ -0,0 +1,926 @@
+    "water_xsdata.set_scatter_matrix(scatter_matrix)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "So far we have not yet set the data for individual materials but have not yet combined them together into a library; we will do that next and then write that library to disk."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "mg_cross_sections_file = openmc.MGXSLibrary(groups)\n",
+    "mg_cross_sections_file.add_xsdatas([uo2_xsdata, mox43_xsdata, mox7_xsdata, mox87_xsdata,\n",
+    "                                    fiss_chamber_xsdata, guide_tube_xsdata, water_xsdata])\n",
+    "\n",
+    "# And finally, write the file to 'mgxs.h5' in the current directory so OpenMC can read it later\n",
+    "mg_cross_sections_file.export_to_hdf5('./mgxs.h5')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Generate 2-D C5G7 Problem Input Files"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To build the actual 2-D model, we will first begin by creating the `materials.xml` file.\n",
+    "\n",
+    "First we need to define materials that will be used in the problem. In other notebooks, either `openmc.Nuclides` or `openmc.Elements` were created at the equivalent stage. We can do that in multi-group mode as well.  However, multi-group cross-sections are sometimes provided as macroscopic cross-sections; the C5G7 benchmark data are macroscopic. In this case, we can instead use `openmc.Macroscopic` objects to in-place of `openmc.Nuclides` or `openmc.Elements`.\n",
+    "\n",
+    "`openmc.Macroscopic`, unlike `openmc.Nuclides` and `openmc.Elements` objects, do not need to be provided enough information to calculate number densities, as no number densities are needed.\n",
+    "\n",
+    "When assigning `openmc.Macroscopic` objects to `openmc.Material` objects, the density can still be scaled by setting the density to a value that is not 1.0. This would be useful, for example, when slightly perturbing the density of water due to a small change in temperature (while of course ignoring any resultant spectral shift). The density of a macroscopic dataset is set to 1.0 in the `openmc.Material` object by default when an `openmc.Macroscopic` dataset is used; so we will show its use the first time and then afterwards it will not be required.\n",
+    "\n",
+    "Aside from these differences, the following code is very similar to similar code in other OpenMC example Notebooks."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Instantiate some Macroscopic Data using a name that is the same as that which was used in the\n",
+    "# MGXS Library earlier\n",
+    "uo2_data = openmc.Macroscopic('uo2')\n",
```

Comment:
Good idea, done.

(but not the cleaner names - the library I am using had the shorter names)